py_ecc/bls/hash.py

```diff
@@ -30,8 +35,31 @@ def hkdf_expand(prk: Union[bytes, bytearray], info: Union[bytes, bytearray], len
         text = previous + info + bytes([i + 1])
 
         # T(i + 1) = HMAC(T(i) || info || i)
-        previous = bytearray(hmac.new(prk, text, hashlib.sha256).digest())
+        previous = bytearray(hmac.new(prk, text, _sha256).digest())
         okm.extend(previous)
 
     # Return first `length` bytes.
     return okm[:length]
+
+
+def sha256(x: bytes) -> bytes:
+    m = _sha256()
+    m.update(x)
+    return m.digest()
+
+
+def xor(a: bytes, b: bytes) -> bytes:
+    return bytes(_a ^ _b for _a, _b in zip(a, b))
```

Comment:
Are you aware that this will truncate the extra characters if either `a, b` is longer than the other?

Reply:
I thought about adding an assert on the difference here, but I didn't want to add the overhead to every signature check. To be fair, the overhead is small in the context of the cost of hashing (dominated by the Fp^2 square root and the co-factor clearing). I only call this function for xor-ing `SHA256` digests. 

Options (not mutually exclusive) for handling this:
* Add in an `assert len(a) == len(b)`
* Rename the function to `_xor`
* zero-pad the shorter case (I am nervous of this behaviour as it is not obvious behaviour for an `xor` function)

Reply:
I mostly just wanted to be sure you were aware of it.  I'll leave it to you to decide whether you think the extra safety is warranted as I'm not clear on how likely it would be for us to erroneously  pass in data that has a mismatched length.

Also, just a reminder that the stdlib has https://docs.python.org/3.5/library/itertools.html#itertools.zip_longest